_posts/2019-08-06-serverless-static-sites.md

```diff
@@ -0,0 +1,232 @@
+
+For this tutorial you will need to install OpenFaaS using an ingress controller instead of exposing the gateway with a LoadBalancer. For this you'll need to set `ingress.enabled=true` and `exposeServices=false` when installing it.
+
+* Install an Ingress Controller
+
+Currently we support [Nginx](https://github.com/kubernetes/ingress-nginx), [Traefik](https://docs.traefik.io/user-guide/kubernetes/) and Zalando's [Skipper](https://opensource.zalando.com/skipper/kubernetes/ingress-controller/). We will be using Nginx for this tutorial:
+```sh
+helm install stable/nginx-ingress --name nginxingress --set rbac.create=true
+```
+
+* [Install the cert-manager](https://docs.openfaas.com/reference/ssl/kubernetes-with-cert-manager/#10-ssl-for-the-gateway) (optional)
+
+Installing the cert-manager to get automatic TLS certificates is optional but I highly recommend it. I will be using it today.
+
+## Install the Ingress Operator
+The Ingress Operator automatically manages the creation of custom ingress rules and TLS certificates for your functions using a new CRD called `FunctionIngress` introduced by Alex in the [previous blog post](https://www.openfaas.com/blog/custom-domains-function-ingress/).  
+Lets deploy the operator to our cluster following the instructions at [the documentation](https://docs.openfaas.com/reference/ssl/kubernetes-with-cert-manager/#20-ssl-and-custom-domains-for-functions):
+```sh
+git clone https://github.com/openfaas-incubator/ingress-operator
+cd ingress-operator
+
+kubectl apply -f ./artifacts/operator-crd.yaml
+kubectl apply -f ./artifacts/operator-rbac.yaml
+kubectl apply -f ./artifacts/operator-amd64.yaml
+```
+
+## Introducing the hugo template
+Today I'm showing you a new template for creating [hugo](https://gohugo.io) static sites that took me little to no time to make thanks to the simplicity and flexibility of [OpenFaaS's templating system](https://www.openfaas.com/blog/template-store/). You can check it out [here](https://github.com/matipan/openfaas-hugo-template).  
+The template copies over the contents of your hugo site, builds it into the `public` directory and then uses a [very lightweight static server](https://gitlab.com/matipan/static-server) that serves the content and provides a healtcheck that follows the standards from OpenFaaS.
+
+## Create a new Hugo site
+We can use the hugo template to create hugo sites and then serve them with custom domain names using OpenFaaS and the Ingress Operator. First lets create the site:
+```sh
+git init
+faas template pull https://github.com/matipan/openfaas-hugo-template
+faas new --lang hugo -g <openfaas gateway url> --prefix <docker hub username> example-site
+```
+This will create a folder called `example-site`, `cd` into it and now create the site with this instructions from the [hugo quickstart guide](https://gohugo.io/getting-started/quick-start/#step-2-create-a-new-site):
+```sh
+hugo new site .
```

Comment:
Where do users install the Hugo CLI? Seems like a pre-req

Reply:
I'll add it above

Reply:
(Did you do this yet?)